examples/solids/qfunctions/finite-strain-neo-hookean-initial-ad.h

```diff
@@ -509,6 +524,26 @@ CEED_QFUNCTION(ElasFSInitialNHdF_AD)(void *ctx, CeedInt Q,
   return 0;
 }
 
+// -----------------------------------------------------------------------------
+// Free tape memory
+// -----------------------------------------------------------------------------
+CEED_QFUNCTION(ElasFSInitialNHFree_AD)(void *ctx, CeedInt Q,
+                                       const CeedScalar *const *in,
+                                       CeedScalar *const *out) {
+  // Inputs
+  void *(*tape) = (void *(*))in[0];
+  // No outputs
+
+  // Quadrature Point Loop
+  CeedPragmaSIMD
+  for (CeedInt i=0; i<Q; i++) {
+    // Free allocated memory for tape
+    free_tape(tape[i]);
+  } // End of Quadrature Point Loop
+
+  return 0;
+}
```

Comment:
@jedbrown 
Here is where the main issue lies. We should instead allocate and free `tape` outside the QFunctions.